src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -342,107 +365,72 @@ static public ServiceInterface create(String configName) {
   // there is no point ... and it just makes it more complicated, if you want to
   // adjust
   // configuration adjust config in the plan before starting
-  static public synchronized ServiceInterface create(String configName, String name, String type) {
-    ServiceInterface si = Runtime.getService(name);
-    if (si != null) {
-      return si;
-    }
+  static public synchronized ServiceInterface create(String name, String type) {

```

Comment:
What are the nullability constraints? Based on other overloads it seems type can be null but name cannot be. There needs to be documentation on what happens when type is null, are there any additional constraints on name in such a case, such as needing to be in the plan?

Reply:
updated